pig/src/org/partiql/pig/main.kt

```diff
@@ -76,7 +78,7 @@ fun generateCode(command: Command.Generate) {
     when (command.target) {
```

Comment:
If you apply the comment I made about adding an `abstract` `domains` property to the `TargetLanguage` `sealed` class, then you can move:

```
val domains = typeUniverse.computeTypeDomains().filterDomains(command.target.domains)
```

Above this line instead of having it 4 times below.

Reply:
(I think)

Reply:
It requires a bit more juggling for the Kotlin language (I'm not sure I like it, but I'll publish it so you can see it).